### PR TITLE
Vaccination

### PR DESCRIPTION
**Pull request by pbkeating**
Added in some extra details on gen_data for vaccination
Need to check the effect of the adding_weights on the number of observations and the age_group

Also found that age_group function created not so helpful age categories when they are yearly age categories, which may be the case for vaccination campaigns e.g. age_group function with breakers = c(0,1,2,3,4) gives age groups 0-0, 1-1,2-2, 3-3, 4-4

### CODE REVIEW

File: inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd [hunk 1]
```diff
@@ -43,83 +43,97 @@ ggplot2::theme_set(theme_bw(base_size = 18))
...
+                             cluster = q77_what_is_the_cluster_number)
+                             
+
+```
 
 
+```{r survey weights}
+######## WEIGHTING 
+
+vaccination_clean <- add_weights(vaccination_clean, population_data_age, age_group, sex)
```
Comment (aspina7) [inline inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd]: the issue was that the age_group variable in the population dataset was not a factor - fixed by using the gen_pop function. 
See this [issue post](https://github.com/R4EPI/sitrep/issues/106#issue-426952648)

File: inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd [hunk 2]
```diff
@@ -43,83 +43,97 @@ ggplot2::theme_set(theme_bw(base_size = 18))
...
+
+
+```{r survey design}
 
 # we first need to create a survey design that reflects the study
 survey_design <- svydesign(
   ids = ~ cluster, # cluster id is stored as cluster
   weights = weights, #sampling weights applied to this example
-  data = vaccine_data_clean
+  data = vaccination_raw_clean
```
Comment (aspina7) [inline inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd]: raw_clean??

File: inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd [hunk 3]
```diff
@@ -164,21 +178,21 @@ plot_age_pyramid(vaccine_data_clean, age_group = "age_group", split_by = "sex")
 Accepting equal validity from self-reported and card-reported vaccination status
 ```{r vaccination coverage overall}
 
-suvery_ciprob(suvery_design, "vaccination_status_simple") %>%
+survey_ciprob(survey_design, "vaccination_status_simple") %>%
```
Comment (aspina7) [inline inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd]: you havent defined vaccination_status_simple yet anywhere?

File: R/gen_data.R [hunk 4]
```diff
@@ -485,6 +485,34 @@ gen_data <- function(dictionary, varnames = "data_element_shortname", numcases =
...
+
+    # cluster number (1 to 30)
+    dis_output$q77_what_is_the_cluster_number <- sample(1:30, numcases, replace = TRUE)
+
+
+    # age in yr (0 to 14) - assuming doing vaccination coverage among those aged less than 15 yrs
+    dis_output$q10_age_yr <- sample(0:14, numcases, replace = TRUE)
+
+    # age in mth (0 to 11)
+    dis_output$q55_age_mth <- sample(0:11, numcases, replace = TRUE)
```
[the remark this answers is not in the document]
Comment (pbkeating) [reply R/gen_data.R]: sounds good

File: inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd [hunk 5]
```diff
@@ -33,7 +33,7 @@ library(epitools) # for creating 2by2 tables
 library(survey)
 
 
-# devtools::install_github("R4EPI/sitrep")
+devtools::install_github("R4EPI/sitrep")
```
[the remark this answers is not in the document]
Comment (pbkeating) [reply inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd]: OK

File: inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd [hunk 6]
```diff
@@ -43,83 +43,97 @@ ggplot2::theme_set(theme_bw(base_size = 18))
...
-vaccine_data <-  vaccine_data_hh %>%
-                left_join(vaccine_data_indiv, by = "fact_0_id")
 
+```
 
-# clean up the column names
-colnames(vaccine_data) <- clean_labels(colnames(vaccine_data))
+```{r read_population_data, warning = FALSE, message = FALSE}
 
+# create fake population by age and sex
```
[the remark this answers is not in the document]
Comment (pbkeating) [reply inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd]: will do

File: inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd (hunk 2 above)
Comment (pbkeating) [reply inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd]: who knows!

File: inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd (hunk 3 above)
Comment (pbkeating) [reply inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd]: Earlier version of the template. Decided on a different angle and haven't update the remaining code yet

File: inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd (hunk 1 above)
Comment (zkamvar) [reply inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd]: I closed the issue. It was a problem with the specification of the population data. One way to ensure all combinations are represented is to use `expand.grid()`

File: inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd [hunk 7]
```diff
@@ -43,83 +43,97 @@ ggplot2::theme_set(theme_bw(base_size = 18))
...
 
-# # read in population data
-# population <- import("your_data",
-#                      which = "sheetName")
+vaccination_raw <- sitrep::gen_data(dictionary = "Vaccination", varnames = "column_name",
+                         numcases = 1000)
 
 
-```
+# Add in information on household number, age and cluster number
```
[the remark this answers is not in the document]
Comment (aspina7) [reply inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd]: deleted.

File: R/gen_data.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (aspina7) [reply R/gen_data.R]: done

File: inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd [hunk 8]
```diff
@@ -43,83 +43,97 @@ ggplot2::theme_set(theme_bw(base_size = 18))
...
+vaccination_raw$q14_hh_no <- sample(1:300, 1000, replace = TRUE)
 
+vaccination_raw$q77_what_is_the_cluster_number <- sample(1:30, 1000, replace = TRUE)
 
-```{r data cleaning}
+vaccination_raw$q10_age_yr <- sample(0:14, 1000, replace = TRUE)
 
-# Need to merge individual data with household data
+vaccination_raw$q55_age_mth <- sample(0:11, 1000, replace = TRUE)
```
[the remark this answers is not in the document]
Comment (aspina7) [reply inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd]: done

**Comment by aspina7** [comment]
To do: 

* [ ] Define vaccine status factor vars 
* [ ] run weighted props w deff for those

**Comment by zkamvar** [comment]
Looks good to me

**Pull request merged**